packages/build-tools/src/steps/functionGroups/submit.ts

```diff
@@ -0,0 +1,601 @@
+
+      echo "$args" | xargs fastlane
+      `,
+    }),
+    createSubmissionEntityFunction().createBuildStepFromFunctionCall(globalCtx, {
+      callInputs: {
+        build_id: buildId,
+        apple_id_username: '${ steps.prepare_asc_api_key.apple_id_username }',
+        asc_app_identifier: '${ steps.prepare_asc_api_key.asc_app_identifier }',
+      },
+    }),
+  ];
+}
+
+function createStepsForAndroidSubmit(
+  {
+    buildId,
+    profile: userProfile,
+    isVerboseFastlaneEnabled,
+    applicationArchivePath: userApplicationArchivePath,
+  }: {
+    buildId: string;
+    profile: string | undefined;
+    isVerboseFastlaneEnabled: boolean | undefined;
+    applicationArchivePath: string | undefined;
+  },
+  globalCtx: BuildStepGlobalContext
+): BuildStep[] {
+  return [
+    createCheckoutBuildFunction().createBuildStepFromFunctionCall(globalCtx),
+    createSetUpNpmrcBuildFunction().createBuildStepFromFunctionCall(globalCtx),
+    createInstallNodeModulesBuildFunction().createBuildStepFromFunctionCall(globalCtx),
+    new BuildStep(globalCtx, {
+      displayName: 'Install additional tools',
+      id: BuildStep.getNewId(),
+      fn: async (stepCtx, { env }) =>
+        await spawnAsync('sudo apt-get -y', ['install', 'jq'], {
+          env,
+          logger: stepCtx.logger,
+          stdio: ['ignore', 'pipe', 'pipe'],
+        }),
+    }),
+    new BuildStep(globalCtx, {
+      displayName: 'Find existing application artifacts',
+      id: 'local_artifacts',
+      outputs: [
+        new BuildStepOutput(globalCtx, {
+          stepDisplayName: 'Find existing application artifacts',
+          id: 'archive_path',
+          required: false,
+        }),
+      ],
+      fn: async (stepCtx, { outputs }) => {
+        if (userApplicationArchivePath) {
+          fs.accessSync(userApplicationArchivePath);
+          stepCtx.logger.info(
+            `Found existing application artifacts: ${userApplicationArchivePath}`
+          );
+          outputs.archive_path.set(userApplicationArchivePath);
+          return;
+        }
+        try {
+          const applicationArchives = await findArtifacts({
+            rootDir: stepCtx.workingDirectory,
+            patternOrPath: 'android/app/build/outputs/**/*.{apk,aab}',
+            logger: stepCtx.logger,
+          });
+
+          if (applicationArchives.length === 0) {
+            return;
+          }
+
+          stepCtx.logger.info(
+            `Found existing application artifacts:\n- ${applicationArchives.join('\n- ')}`
+          );
+
+          if (applicationArchives.length > 1) {
+            throw new Error(
+              'Found more than one application archive. Provide `application_archive_path` input.'
+            );
+          }
+          outputs.archive_path.set(applicationArchives[0]);
+        } catch (err: any) {
+          stepCtx.logger.info({ err }, `Failed to find existing application artifacts.`);
+        }
+      },
+    }),
```

Comment:
I believe we can move it to a function we can reuse for both Android and iOS or just generally `eas/` custom function